lib/routes/fiat_currencies/fiat_currency_settings.dart

```diff
@@ -0,0 +1,274 @@
+      _preferredFiatCurrencies =
+          List.from(user.fiatCurrencyPreferences.preferredFiatCurrencies);
+    });
+  }
+
+  void _initializeFiatCurrencyList() {
+    _accountBloc.accountStream
+        .firstWhere((account) => account.fiatConversionList.isNotEmpty)
+        .then((account) {
+      _fiatConversionList = List.from(account.fiatConversionList);
+      _accountModel = account;
+      _sortList();
+    });
+  }
+
+  void _sortList() {
+    _updateSelectedFiatConversions();
+    _unselectedFiatConversions = List.from(_fiatConversionList.where(
+        (fiatConversion) => !_preferredFiatCurrencies
+            .contains(fiatConversion.currencyData.shortName)));
+    _sortByOrder();
+    _sortByAlphabet();
+    _fiatConversionList = _selectedFiatConversions + _unselectedFiatConversions;
+  }
+
+  void _sortByOrder() {
+    Map<String, int> order = new Map.fromIterable(_preferredFiatCurrencies,
+        key: (key) => key,
+        value: (key) => _preferredFiatCurrencies.indexOf(key));
+    _selectedFiatConversions.sort((a, b) => order[a.currencyData.shortName]
+        .compareTo(order[b.currencyData.shortName]));
+  }
+
+  void _sortByAlphabet() {
+    _unselectedFiatConversions.sort((a, b) => a.currencyData.shortName
+        .toString()
+        .compareTo(b.currencyData.shortName.toString()));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return WillPopScope(
+      onWillPop: () => _onWillPop(),
+      child: Scaffold(
+        appBar: AppBar(
+          iconTheme: Theme.of(context).appBarTheme.iconTheme,
+          textTheme: Theme.of(context).appBarTheme.textTheme,
+          backgroundColor: Theme.of(context).canvasColor,
+          leading: backBtn.BackButton(
+            onPressed: () => _onWillPop(),
+          ),
+          title: Text(
+            "Fiat Currencies",
+            style: Theme.of(context).appBarTheme.textTheme.headline6,
+          ),
+          elevation: 0.0,
+        ),
+        body: StreamBuilder<AccountModel>(
+            stream: _accountBloc.accountStream,
+            builder: (context, snapshot) {
+              AccountModel account = snapshot.data;
+              if (!snapshot.hasData) {
+                return Container();
+              }
+
+              if (account.fiatConversionList.isEmpty ||
+                  account.fiatCurrency == null) {
+                return Loader();
+              }
+
+              return DragAndDropList(
+                _fiatConversionList.length,
+                itemBuilder: (BuildContext context, index) =>
+                    _buildFiatCurrencyTile(_fiatConversionList[index], index),
+                onDragFinish: _onReorder,
+                canDrag: (index) => index < _selectedFiatConversions.length,
+                canBeDraggedTo: (from, to) =>
+                    to < _selectedFiatConversions.length,
+                dragElevation: 8.0,
+              );
+            }),
+      ),
+    );
+  }
+
+  _onWillPop() async {
```

Comment:
Usually users either have "Save" button to save changes or the changes are saved as they edit.
Here the save happens implicitly when user click "Back". I think it is better to save changes on each change and remove this code.
One side effect with the current code is that user gets different behaviour when pressing the back menu and when pressing the back android button.

Reply:
[Done](https://github.com/breez/breezmobile/pull/394/commits/1381bcf8bcb3e0a35d8f3893f8d8057de37a08bc)